gpbft/api.go

```diff
@@ -51,13 +51,26 @@ type MessageReceiver interface {
 	ReceiveAlarm() error
 }
 
+// Supports catching-up with other participants by pulling finality certificates.
+type FinalityReceiver interface {
```

Comment:
I think this name will be confusing, given that this module is the thing that usually produces finality decisions (see DecisionReceiver, which is implemented by the host).

But also, I don't think we need to pass or validate any information here other than the instance to skip to. The participant will then turn around and fetch the power table, beacon etc needed for that instance when it starts. I don't think this method or its documentation should reference finality certificates or anything else about how the host decided to move to this instance: just do what it's told.

I think we can just inline this in Receiver as a method like `SkipToInstance(instance)`. In fact, if the initial instance was a parameter to `Start` rather than a configuration option, we could merge them into one method. We should consider removing `Start()` once finished here.